File: Erikti-frontend/src/app/services/models/volunteer.ts
import {User} from "./user";

export interface Volunteer {
  volunteerId?: number;
  user?: User;
  firstName?: string;
  lastName?: string;
  email?: string;
  phone?: string;
  address?: string;
  city?: City;
  aboutMe?: string;
  birthday?: string;
  gender?: string;
  experienceMonth?: number;
  rating?: number;
}


export enum City {
  ASTANA = 'ASTANA',
  ALMATY = 'ALMATY',
  KYZYLORDA = 'KYZYLORDA',
  TARAZ = 'TARAZ',
  SHYMKENT = 'SHYMKENT',
  AKTAU = 'AKTAU',
  SEMEI = 'SEMEI',
  OSKEMEN = 'OSKEMEN',
  AKTOBE = 'AKTOBE',
  PAVLODAR = 'PAVLODAR',
  TALDYKORGAN = 'TALDYKORGAN',
  KOKSHETAU = 'KOKSHETAU',
  KOSTANAY = 'KOSTANAY',
  ATYRAY = 'ATYRAY',
  TURKISTAN = 'TURKISTAN',
  PETROPAVL = 'PETROPAVL',
  ZHEZKAZGAN = 'ZHEZKAZGAN',
  ORAL = 'ORAL'
}

File: Erikti-backend/src/main/java/kz/erasyl/volunteerback/models/Volunteer.java
package kz.erasyl.volunteerback.models;

import com.fasterxml.jackson.annotation.*;
import jakarta.persistence.*;
import kz.erasyl.volunteerback.models.enums.City;
import lombok.*;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;


@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@ToString
@Table(name = "volunteer")
public class Volunteer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "volunteerId")
    private Long volunteerId;

    @OneToOne(fetch = FetchType.EAGER)
    private User user;


    private String firstName;
    private String lastName;
    private String email;
    private String phone;
    private String address;

    @Enumerated(EnumType.STRING)
    private City city;
    private String aboutMe;

    private LocalDate birthday;
    private String gender;
    private Integer experienceMonth;
    private Float rating;
    private Integer numberOfRates;
    private Boolean banned;

    @OneToMany(mappedBy = "volunteer")
    @JsonIgnore
    Set<VolunteerEventRegistration> registrations;

    @OneToMany(mappedBy = "volunteer")
    @JsonIgnore
    private Set<VolunteerOrganizationRating> volunteerOrganizationRating;

//        @ManyToMany(mappedBy = "volunteers")
//        @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "volunteerId")
//
//        private List<Event> events;



}
